jupyterhub/tests/selenium/test_browser.py

```diff
@@ -209,36 +225,686 @@ async def test_open_url_login(
+        await asyncio.sleep(0.001)
+    while '/spawn-pending/' in browser.current_url and is_displayed(
+        browser, SpawningPageLocators.PROGRESS_BAR
+    ):
+        # checking text messages that the server is starting to up
+        texts = browser.find_elements(*SpawningPageLocators.TEXT_SERVER)
+        texts_list = []
+        for text in texts:
+            text.text
+            texts_list.append(text.text)
+        assert str(texts_list[0]) == SpawningPageLocators.TEXT_SERVER_STARTING
+        assert str(texts_list[1]) == SpawningPageLocators.TEXT_SERVER_REDIRECT
+        # checking progress of the servers starting (messages, % progress and events log))
+        logs_list = []
+        # await asyncio.sleep(0.001)
+        try:
+            progress_bar = browser.find_element(*SpawningPageLocators.PROGRESS_BAR)
+            progress = browser.find_elements(*SpawningPageLocators.PROGRESS_MESSAGE)
+            logs = browser.find_elements(*SpawningPageLocators.PROGRESS_LOG)
+            for status_message in progress:
+                progress_messages = status_message.text
+                percent = (
+                    progress_bar.get_attribute('style')
+                    .split(';')[0]
+                    .split(':')[1]
+                    .strip()
+                )
+                for i in range(len(logs)):
+                    progress_log = logs[i].text
+                    logs_list.append(progress_log)
+                if progress_messages == "":
+                    assert percent == "0%"
+                    assert int(len(logs_list)) == 0
+                if progress_messages == "Server requested":
+                    assert percent == "0%"
+                    assert int(len(logs_list)) == 1
+                    assert str(logs_list[0]) == "Server requested"
+                if progress_messages == "Spawning server...":
+                    assert percent == "50%"
+                    assert int(len(logs_list)) == 2
+                    assert str(logs_list[0]) == "Server requested"
+                    assert str(logs_list[1]) == "Spawning server..."
+                if "Server ready at" in progress_messages:
+                    assert (
+                        f"Server ready at {app.base_url}user/{user}/"
+                        in progress_messages
+                    )
+                    assert percent == "100%"
+                    assert int(len(logs_list)) == 3
+                    assert str(logs_list[0]) == "Server requested"
+                    assert str(logs_list[1]) == "Spawning server..."
+                    assert (
+                        str(logs_list[2])
+                        == f"Server ready at {app.base_url}user/{user}/"
+                    )
+                    assert str(logs_list[2]) == progress_messages
+        except NoSuchElementException:
+            break
+        except StaleElementReferenceException:
```

Comment:
This can be one except with multiple exceptions to catch:

```suggestion
        except (NoSuchElementException, StaleElementReferenceException):
```